src/token/components/TokenSelector.tsx

```diff
@@ -0,0 +1,58 @@
+import { TokenSelectorReact } from '../types';
+import { TokenImage } from './TokenImage';
+
+function CaretDown() {
+  return (
+    <svg width="16" height="17" viewBox="0 0 16 17" fill="none" xmlns="http://www.w3.org/2000/svg">
+      <path
+        d="M12.95 4.85999L8.00001 9.80999L3.05001 4.85999L1.64001 6.27999L8.00001 12.64L14.36 6.27999L12.95 4.85999Z"
+        fill="#0A0B0D"
+      />
+    </svg>
+  );
+}
+
+const styles = {
+  button: {
+    borderRadius: '16px',
+    padding: '4px 12px 4px 12px',
+    display: 'flex',
+    alignItems: 'center',
+    background: '#EEF0F3',
+    width: 'fit-content',
+  },
+  image: {
+    height: '16px',
+    width: '16px',
+  },
+  label: {
+    color: '#0A0B0D',
+    fontSize: '16px',
+    lineHeight: '1.5',
+    fontWeight: '500',
+    marginLeft: '4px',
+  },
+  caret: {
+    marginLeft: '8px',
+  },
+};
+
+export function TokenSelector({ token, onClick }: TokenSelectorReact) {
```

Comment:
I am very tempted to merge TokenChip into TokenSelector..... what do we think? How other design system offer this? 🤔

Reply:
i had the same thought but the functionality is so different between the two

Reply:
TokenSelector to me feels more a super set of a TokenChip. I guess it's cleaner as component. But at the same time they both end up delegating the `onClick` experience to the dev, to decide.

So in the end the question we ask is not even if you want or not `onClick` but more, do you want a Selector image yes or no.

Ok so TokenChip is pretty much a tag component (https://ant.design/components/tag), where TokenSelector is a select component (https://ant.design/components/select).

I think at first they for sure are very similar, BUT, if we end up building the Tag experience, and Select experience like other Design System do. 

Then yeah, I think keep them split makes-sense for the long-term.

Reply:
Had similar thoughts. I did initially extend TokenChip with postfix icon but it didn't feel right.

I could see a world where there is multiple options for TokenSelector. Pop up the big token selector component with chips, row and search or just a simple dropdown.